enginetest/queries/information_schema_queries.go

```diff
@@ -1451,6 +1451,29 @@ from information_schema.routines where routine_schema = 'mydb' and routine_type
 			},
 		},
 	},
+	{
+		// this test verifies that https://github.com/dolthub/dolt/issues/5549 is fixed
+		Name: "correct schema for a recursive cte that calculate fibonacci numbers",
```

Comment:
The error happens outside of a view, but we need to get the schema of the results, so the view is a simple way of verifying the output types.
Tests in `queries.go` ignore the output schema, so they don't catch/trigger this bug.

Since the bug is literally "recursive CTEs have the wrong schema", I thought this spot would be the correct option. (But keep in mind [this bug](https://github.com/dolthub/dolt/issues/3168) that's blocking us from checking the schema in the same way that we check other cases.) If there's a better place for "is FOO returning the correct types" tests, I'll move it there.

Reply:
`queries.go` uses `QueryTest` structs, which have an option to validate result schemas

Reply:
Cool, moving.